src/main/java/spoon/support/StandardEnvironment.java

```diff
@@ -631,4 +633,14 @@ public PrettyPrinter createPrettyPrinter() {
 	public void setPrettyPrinterCreator(Supplier<PrettyPrinter> creator) {
 		this.prettyPrinterCreator = creator;
 	}
+
+	@Override
+	public boolean isAllowMultiTypeDefinitions() {
+		return allowMultiTypeDefinitions;
+	}
+
+	@Override
+	public void setAllowMultiTypeDefinitions(boolean allowMultiTypeDefinitions) {
+		this.allowMultiTypeDefinitions = allowMultiTypeDefinitions;
```

Comment:
No it does not work like that since it is the same signature the model should correct (at least in the type level).
It is basically the same behaviour as the runtime environment of java. When the same class is declared several times only one is loaded is considered.
It is not perfect because people do dirty stuff that but it is unrelated to noclasspath.

Reply:
the signature of the class is the same, but inside the class you can have different members, can't you?
Like in your example, what happens if you have one A with method `foo()` and another with `bar()` and let's say you want to spoon the model to check invocations of `bar()`?

Reply:
Do you realize that this would completely break the execution of the application?

Reply:
pretty much like when you don't load the right logger implementation yes. :) 
I don't know your usecase here, maybe you're trying to load multiple version of the same project with small differences between them, so it'll be wrong in that case.

Reply:
ok. I think a better name can be good for the option then: I misunderstood its goals at first :) 
Something like `allowIdenticalDeclarations`? WDYT?